checks/common.go

```diff
@@ -0,0 +1,21 @@
+package checks
+
+import (
+	v1 "github.com/flanksource/canary-checker/api/v1"
+	"github.com/flanksource/kommons"
+)
+
+func GetAuthValues(auth *v1.Authentication, client *kommons.Client, namespace string) (username, password string, err error) {
```

Comment:
I would return a new copy of Auth with the values filled in - This would allow future expansion to server/client-side TLS where supported etc...